auth/src/main/java/com/firebase/ui/auth/provider/FacebookProvider.java

```diff
@@ -182,9 +201,32 @@ private IdpResponse createIdpResponse(String email, LoginResult loginResult) {
                 loginResult.getAccessToken().getToken());
     }
 
-    private void onSuccess(String email, LoginResult loginResult) {
-        gcCallbackManager();
-        mCallbackObject.onSuccess(createIdpResponse(email, loginResult));
+    private void onSuccess(final String email, final LoginResult loginResult) {
+        if (sPublishPermissions.isEmpty()) {
+            gcCallbackManager();
+            mCallbackObject.onSuccess(createIdpResponse(email, loginResult));
+        } else {
+            LoginManager loginManager = LoginManager.getInstance();
+            loginManager.registerCallback(sCallbackManager, new FacebookCallback<LoginResult>() {
+                @Override
+                public void onSuccess(LoginResult result) {
+                    sPublishPermissions.clear();
+                    FacebookProvider.this.onSuccess(email, loginResult);
+                }
+
+                @Override
+                public void onCancel() {
+                    FacebookProvider.this.onCancel();
+                }
+
+                @Override
+                public void onError(FacebookException error) {
+                    FacebookProvider.this.onError(error);
+                }
+            });
+            loginManager.logInWithPublishPermissions((Activity) mCallbackObject,
```

Comment:
I agree with you.  Facebook's guidelines say not to request publish permissions at the time of the first login and that's all we really handle.  If this had turned out super easy and clean I would have said we can put it in, but not worth the trouble!

Thanks for investigating.